src/shared/components/FluxMonacoEditor.tsx

```diff
@@ -130,36 +134,41 @@ const FluxEditorMonaco: FC<Props> = ({
     onChangeScript(text)
   }
 
-  return (
-    <ErrorBoundary>
-      <div className={wrapperClassName} data-testid="flux-editor">
-        <MonacoEditor
-          language={FLUXLANGID}
-          theme={THEME_NAME}
-          value={script}
-          onChange={onChange}
-          options={{
-            fontSize: 13,
-            fontFamily: '"IBMPlexMono", monospace',
-            cursorWidth: 2,
-            lineNumbersMinChars: 4,
-            lineDecorationsWidth: 0,
-            minimap: {
-              renderCharacters: false,
-            },
-            overviewRulerBorder: false,
-            automaticLayout: true,
-            readOnly: readOnly || false,
-            wordWrap: wrapLines ?? 'off',
-            scrollBeyondLastLine: false,
-          }}
-          editorDidMount={editorDidMount}
-        />
-        <div id={ICON_SYNC_ID} className="sync-bar">
-          <Icon glyph={IconFont.Sync} className="sync-icon" />
+  return useMemo(
```

Comment:
Added use `useMemo()` because the tests were failing otherwise.

Review the last commit, before and after the change. We had the browser re-rendering for 1 extra cycle per keydown (key typed) in the editor. This meant that in the tests with the super fast typing, it did not always correctly perform the recognition of the cypress-typed closing paran (versus the auto-complete paran provided earluer). Therefore we were failing with extra closing brackets/parans.

Example failures (from CI test artifacts):
<img width="598" alt="Screen Shot 2022-09-06 at 3 08 52 PM" src="https://user-images.githubusercontent.com/10232835/188756241-a2dd7fe7-0bfc-4b3b-9357-3e4faff7d835.png">
<img width="597" alt="Screen Shot 2022-09-06 at 3 07 45 PM" src="https://user-images.githubusercontent.com/10232835/188756286-173055c4-8c7e-457c-90fd-311c1da6a488.png">

Reply:
Good to know!